tracer/src/Datadog.Trace/ContinuousProfiler/ContextTracker.cs

```diff
@@ -0,0 +1,120 @@
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ContextTracker));
+
+        private readonly ProfilerStatus _status;
+        private readonly bool _isFlagSet;
+        private readonly ThreadLocal<IntPtr> _traceContextPtr;
+
+        public ContextTracker(ProfilerStatus status)
+        {
+            _status = status;
+            _isFlagSet = EnvironmentHelpers.GetEnvironmentVariable(EnvironmentVariables.CodeHotspotEnabled)?.ToBoolean() ?? false;
+            _traceContextPtr = new ThreadLocal<IntPtr>();
+            Log.Information("CodeHotspots feature is {IsEnabled}.", _isFlagSet ? "enabled" : "disabled");
+        }
+
+        public bool IsEnabled
+        {
+            get
+            {
+                return _status.IsProfilerReady && _isFlagSet;
+            }
+        }
+
+        public void Set(ulong localRootSpanId, ulong spanId)
+        {
+            WriteToNative(new SpanContext(localRootSpanId, spanId));
+        }
+
+        public void Reset()
+        {
+            WriteToNative(SpanContext.Zero);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private static void WriteContext(IntPtr ptr, in SpanContext ctx)
+        {
+            Marshal.StructureToPtr(ctx, ptr, false);
+        }
+
+        private void EnsureIsInitialized()
+        {
+            if (_traceContextPtr.IsValueCreated)
+            {
+                return;
+            }
+
+            try
+            {
+                _traceContextPtr.Value = NativeInterop.GetTraceContextNativePointer();
+            }
+            catch (Exception e)
+            {
+                Log.Debug("Unable to get the tracing context pointer for the thread {ThreadID}: {Reason}", Environment.CurrentManagedThreadId.ToString(), e.Message);
+                _traceContextPtr.Value = IntPtr.Zero;
+            }
+        }
+
+        private void WriteToNative(in SpanContext ctx)
+        {
+            if (!IsEnabled)
+            {
+                return;
+            }
+
+            EnsureIsInitialized();
+
+            var ctxPtr = _traceContextPtr.Value;
+
+            if (ctxPtr == IntPtr.Zero)
+            {
+                return;
+            }
+
+            try
+            {
+                WriteContext(ctxPtr, ctx);
+            }
+            catch (Exception e)
+            {
+                Log.Debug("Failed to write tracing context at {CtxPtr} for {ThreadID}: {Reason}", ctxPtr, Environment.CurrentManagedThreadId.ToString(), e.Message);
+            }
+        }
+
+        // This stuct in defined in <repos>/profiler/src/ProfilerEngine/Datadog.Profiler.Native/ManagedThreadInfo.h
+        // /!\ We must keep both layout in sync.
+        [StructLayout(LayoutKind.Explicit)]
+        private readonly struct SpanContext
+        {
+            public static readonly SpanContext Zero = new(0, 0);
+
+            [FieldOffset(0)]
+            public readonly ulong LocalRootSpanId;
```

Comment:
I just want to make sure that what you want really is the root span id and not the trace id. In some tracers, they happen the be the same value, but this is not required and in .NET they are not the same.

Reply:
I confirm it's the root span id